### ISSUE TITLE
hotkey: "Cycle through the available video tracks"

### ISSUE THREAD

**Issue by BETLOG** [mpv-player/mpv#9577]
hotkey: '_' Cycle through the available video tracks. - makes MPV completely vanish.

- mpv version
mpv --version
mpv 0.32.0 Copyright © 2000-2020 mpv/MPlayer/mplayer2 projects
 built on UNKNOWN
ffmpeg library versions:
   libavutil       56.31.100
   libavcodec      58.54.100
   libavformat     58.29.100
   libswscale      5.5.100
   libavfilter     7.57.100
   libswresample   3.5.100
ffmpeg version: 4.2.4-1ubuntu0.1

- Linux Distribution and Version
cat /var/log/installer/media-info 
Kubuntu 20.04.1 LTS "Focal Fossa" - Release amd64 (20200731)
 
- Source of the mpv binary
default repo
- If known which version of mpv introduced the problem
- Window Manager and version
update-alternatives --list x-window-manager
/usr/bin/kwin_x11
cat /etc/X11/default-display-manager
/usr/bin/sddm
 
- GPU driver and version
cat /proc/driver/nvidia/version
NVRM version: NVIDIA UNIX x86_64 Kernel Module  460.91.03  Fri Jul  2 06:04:10 UTC 2021
GCC version:  gcc version 9.3.0 (Ubuntu 9.3.0-17ubuntu1~20.04) 

### Reproduction steps
Play video with more than 1 video stream.
Push "_" (Cycle through the available video tracks.) several times

### Expected behavior
stream 1 plays
Push "_" (Cycle through the available video tracks.)
second video stream is now visible
Push "_" (Cycle through the available video tracks.)
There is no 3rd video stream SO FIRST STREAM IS VISIBLE AGAIN

### Actual behavior
Play video with more than 1 video stream.
video stream 1 plays
Push "_" (Cycle through the available video tracks.)
second video stream is now visible
Push "_" (Cycle through the available video tracks.)
If there is no 3rd video stream - MPV vanishes but is still running and producing audio. Has to be killed manually because it is no longer focused/visible and hotkeys do not work.

### Log file
https://pastebin.com/3CuFtf16

**Comment by guidocella** [comment]
You can still press `_` in the terminal where mpv is running, otherwise add `force-window` to `~/.config/mpv/mpv.conf` so there will still be an empty window that can receive `_` when no video track is selected.

**Comment by BETLOG** [comment]
Workarounds are good. I'll see which I prefer in practice :)

But given that there is no "decrement video stream" and the word "cycle" is in the hotkey description (and "increment video stream into the void" is not ) I would still suggest that the current behaviour is incorrect.

**Comment by guidocella** [comment]
The `cycle` command cycles through all the values a property or option can assume. `vid=no` is one of the possible values for `vid`, and it's useful when you want to hide the window and only listen to the audio.

**Comment by BETLOG** [comment]
Valid.
But still a bit counter-intuitive if you arent running it from a terminal and have to manually kill the windowless task simply because of cycling one too many video streams.

Ok this is odd: I can no longer get the window to vanish. Instead it always shows the 'no' index.
Even when I am *not* starting it with --force-window
Is this a persistent setting after being invoked on the command line once? mpv.conf does not exist

**Comment by guidocella** [comment]
If you aren't running mpv from a terminal, it automatically enables the pseudo GUI mode which sets `force-window=yes`. See https://mpv.io/manual/master/#pseudo-gui-mode.

**Comment by BETLOG** [comment]
This is unusual. Now that I have used --force-window *once* it is now behaving as expected. ie: it is displaying the 'no' window, which it has never done before.... without using --force-window=yes

Previously: 
- When started by right clicking an mkv/mp4 file in Dolphin, then mpv -> press _ several times = mpv window eventually vanishes 
- Same as above when started by a script i use to generate screenshots from which i extract millisecond-accurate timestamps:
-(there is no terminal used here, nor invoked by the .desktop script handler)
```
    cmd="mpv "
    cmd+="--volume=50 "
    cmd+="--fullscreen --loop-file "
    cmd+="--screenshot-format=png "
    cmd+="--screenshot-template=\"${input%.*}_timestamp_%P\" "
    [[ -n $startAt ]] && cmd+="--start=$startAt "
    cmd+="'$input'"
    echo "$cmd"
    eval $cmd
```

But now: It displays the 'no' window regardless of how I start mpv. Which in this case is what I want.
However it was not until I first started mpv from the command line with --force-window=yes that this behaviour changed.

So to summarise: it now behaves as expected, regardless of omitting --force-window=yes,  but for no apparent reason.
...Unless there is a config file that I am unaware of which is storing --force-window as a general preference

**Comment by avih** [comment]
Closing, assuming pseudo-gui mode handles it correctly.

**Comment by elraymond** [comment]
Try, for example,
`_  set force-window yes; cycle video`
in input.conf . That way it sets the force-window option only when you do actually cycle through the video tracks, and playback of pure audio files isn't affected.

**Comment by BETLOG** [comment]
Noted. But whatever utility i was scripting is written now, and i'm over it. So I'll probably never know if that helps :)
